File: packages/ui/src/schemas/audit.ts
import { z } from 'zod';

export const AuditLogSchema = z.object({
  id: z.string(),
  timestamp: z.string(),
  user_id: z.string().nullable(),
  user_email: z.string().nullable(),
  user_role: z.string().nullable(),
  action: z.string(),
  resource_type: z.string(),
  resource_id: z.string().nullable(),
  details: z.record(z.unknown()).nullable(),
  ip_address: z.string().nullable(),
  user_agent: z.string().nullable(),
});

export const AuditLogListSchema = z.object({
  items: z.array(AuditLogSchema),
  total: z.number(),
});

export type AuditLog = z.infer<typeof AuditLogSchema>;
export type AuditLogList = z.infer<typeof AuditLogListSchema>;

File: packages/api/src/schemas/audit.py
"""
Audit log Pydantic schemas.
"""

from datetime import datetime

from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    """Single audit log entry."""

    id: str
    timestamp: datetime
    user_id: str | None = None
    user_email: str | None = None
    user_role: str | None = None
    action: str
    resource_type: str
    resource_id: str | None = None
    details: dict | None = None
    ip_address: str | None = None
    user_agent: str | None = None

    model_config = {"from_attributes": True}


class AuditLogListResponse(BaseModel):
    """Paginated audit log list."""

    items: list[AuditLogResponse]
    total: int
